OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,421 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds an overlay for weather animations like falling DroppingParticles or raindrops")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Can be true or false")]
+		public readonly bool WeatherEnabled = true;
+		[Desc("Can be snow or rain or d2kSand")]
+		public readonly string WeatherVariants = "snow";
+		[Desc("Can be true or false")]
+		public readonly bool Wind = true;
+		[Desc("Can be true or false")]
```

Comment:
The purpose of these descriptions is to describe what they do (what is InstantWind?) - the fact that they can be true or false is obvious from them being bools.